src/release/breaking-changes/boxdecoration-shape-migration.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Migration guide for BoxDecoration.circle
+description: Removal of BoxDecoration.circle and how to migrate
+---
+
+## Summary
+
+This change removes `BoxDecoration(shape: BoxShape.circle)`. Existing apps
+should use `ShapeDecoration(shape: CircleBorder())` instead.
+
+## Context
+
+BoxDecoration.shape uses an enum. Each new enum value requires new code
+in a bunch of places. That new code likely won't get tree-shaken out,
+since the compiler doesn't know which values are never used.
+This has a permanent binary size cost even for people who don't use it.
+Flutter moved to ShapeBorders rather than BoxShape for new
+shapes, and the existence of BoxShape is just a legacy feature.
```

Comment:
**suggestions:** Tighten up the language. This is not required, just recommended.

```suggestion
in many places. That new code may not shake out,
since the compiler doesn't know which values never get used.
`BoxDecoration` has a permanent binary size cost,
even for people who don't use it.
Flutter moved to `ShapeBorders` rather than `BoxShape` for new
shapes, and the existence of `BoxShape` remains as a legacy feature.
```